### PR TITLE
Androapp 4953

### PR DESCRIPTION
**Pull request by mvmmacamo2**
## Description
Please include a summary of the change and include the related jira issue if it exists.

[ jira issue ](https://dhis2.atlassian.net/browse/ANDROAPP-4953

## Solution description
1. Write method to call Dialog
2. implement interface to listen Dialog data Change
3. Pass data to compose class UI 
4. Create Method in ViewModel to listen the Dialog data change
5. Set Data to OutlinedTextField
6. Test functinalities. 

## Covered unit test cases
Describe the tests that you ran to verify your changes.
## Where did you test this issue?
- [ x] Smartphone Emulator
- [ ] Tablet Emulator
- [ ] Smartphone
- [ ] Tablet
## Which Android versions did you test this issue?
- [ ] 4.4
- [ ] 5.X - 6.X
- [ ] 7.X
- [ ] 8.X
- [ ] 9.X - 10.X
- [ x] Other
## Checklist
- [ x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -27,6 +27,7 @@ plugins {
     id "org.jlleitschuh.gradle.ktlint" version "9.2.1"
     id "org.sonarqube" version "3.3"
     id 'org.jetbrains.kotlin.jvm' version '1.6.10' apply false
+    id 'org.jetbrains.kotlin.android' version '1.7.10' apply false
```
Comment (andresmr) [inline build.gradle:30]: why do you need this?

File: stock-managment/src/main/java/org/dhis2/android/rtsm/ui/home/HomeViewModel.kt [hunk 2]
```diff
@@ -80,6 +80,9 @@ class HomeViewModel @Inject constructor(
     private val _deliveryTo = MutableStateFlow<String?>(null)
     val deliveryTo: StateFlow<String?> get() = _deliveryTo
 
+    private val _test = MutableStateFlow("")
+    val test: StateFlow<String> get() = _test
```
Comment (andresmr) [inline stock-managment/src/main/java/org/dhis2/android/rtsm/ui/home/HomeViewModel.kt:84]: I don't understand the purpose of this property, could you add a more meaningful name?

**Review by andresmr — CHANGES_REQUESTED** [review]

File: stock-managment/src/main/java/org/dhis2/android/rtsm/ui/home/HomeViewModel.kt (hunk 2 above)
Comment (mvmmacamo2) [reply stock-managment/src/main/java/org/dhis2/android/rtsm/ui/home/HomeViewModel.kt]: This as been solved.

File: commons/src/main/java/org/dhis2/commons/orgunitdialog/CommonOrgUnitDialog.java [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+        instance = null;
+        isMultiSelection = false;
+        possitiveListener = null;
+        negativeListener = null;
+        title = null;
+        myOrgs = null;
+        mySelectedOrg = null;
+        selectedOrg = null;
+
+
```
Comment (andresmr) [inline commons/src/main/java/org/dhis2/commons/orgunitdialog/CommonOrgUnitDialog.java:69]: could you clean these spaces?

File: commons/src/main/java/org/dhis2/commons/utils/OrgUnitHolder_2.java [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+import java.util.Locale;
+import java.util.Objects;
+
+import timber.log.Timber;
+
+/**
+ * QUADRAM. Created by ppajuelo on 19/10/2017.
+ */
+
+public class OrgUnitHolder_2 extends TreeNode.BaseNodeViewHolder<OrganisationUnit> {
```
Comment (andresmr) [inline commons/src/main/java/org/dhis2/commons/utils/OrgUnitHolder_2.java:26]: why is _2? Could not be ORgUnitHolder?

Comment (mvmmacamo2) [reply commons/src/main/java/org/dhis2/commons/utils/OrgUnitHolder_2.java:26]: Because already exist a file with tthis name (ORgUnitHolder)

**Review by andresmr — APPROVED** [review]

**Review by andresmr — APPROVED** [review]

**Review by andresmr — APPROVED** [review]

**Pull request merged**